app/application/use_cases/request_otp.py

```
@@ -0,0 +1,43 @@
+    ):
+        self._otp_service = otp_service
+        self._email_service = email_service
+        self._rate_limiter = rate_limiter
+
+    async def execute(self, data: RequestOTPInput) -> None:
+        if not await self._rate_limiter.allow_request(
+            key=f"otp_req:{data.client_ip}",
+            limit=settings.RATE_LIMIT_OTP_REQ_PER_MIN,
+            window_seconds=60
+        ):
+            raise RateLimitError("Too many requests from this IP")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)